File: Python Fundamentals/Mid Exam problems/shopping_list.py
initial_list = input().split('!')

while True:
    command = input()
    if command == "Go Shopping!":
        print(', '.join(initial_list))
        break

    command_list = command.split()
    if command_list[0] == 'Urgent':
        if command_list[1] not in initial_list:
            initial_list.insert(0, command_list[1])

    elif command_list[0] == 'Unnecessary':
        if command_list[1] in initial_list:
            initial_list.remove(command_list[1])

    elif command_list[0] == 'Correct':
        if command_list[1] in initial_list:
            initial_list[initial_list.index(command_list[1])] = command_list[2]

    elif command_list[0] == 'Rearrange':
        if command_list[1] in initial_list:
            initial_list.remove(command_list[1])
            initial_list.append(command_list[1])

File: JavaScript Fundamentals/Mid Exams/Mid Exam 4/shoppingList.js
function solve(arr) {
    let groceries = arr.shift().split('!');
    for (let command of arr) {
        if (command == "Go Shopping!") break;
        let tokens = command.split(' ');
        if (tokens[0] == 'Urgent') {
            if (!(groceries.includes(tokens[1]))) groceries.splice(0, 0, tokens[1]);
        } else if (tokens[0] == 'Unnecessary') {
            if (groceries.includes(tokens[1])) groceries.splice(groceries.indexOf(tokens[1]), 1);
        } else if (tokens[0] == 'Correct') {
            if (groceries.includes(tokens[1])) { 
                groceries[groceries.indexOf(tokens[1])] = tokens[2];
            }
        } else if (tokens[0] == 'Rearrange') {
            if (groceries.includes(tokens[1])) {
                groceries.splice(groceries.indexOf(tokens[1]), 1);
                groceries.push(tokens[1]);
            }
        }
    }
    console.log(groceries.join(', '));
}
